docs/HYSequenceDiagram.puml

```diff
@@ -0,0 +1,49 @@
+
+@startuml
+!include Style.puml
+'https://plantuml.com/sequence-diagram
+hide footbox
+
+
+
+
+
+participant "<<class>>\nParserUtil" as PU COL4
+
+-> PU : .parseDate("20000101")
+activate PU
+
+participant "<<class>>\nDate" as DATE COL2
+
+PU -> DATE : isValidDate(\n"20000101")
+activate DATE COL2
+
+DATE -> DATE : produceValidationFormat("20000101")
+activate DATE COL2
+DATE --> DATE : "yyyyMMdd"
+deactivate DATE COL2
+DATE --> PU
+deactivate DATE COL2
+create ":Date" as D COL1
+
+PU -> D
+activate D COL1
+
+D -> DATE : isValidDate("20000101")
+activate DATE COL2
+DATE -> DATE : produceValidationFormat("20000101")
+activate DATE COL2
+DATE --> DATE : "yyyyMMdd"
+deactivate DATE COL2
+DATE --> D
+deactivate DATE COL2
+D -> D : checkArgument(validity, constraints)
```

Comment:
I'm not sure if showing the process of calling checkArgument is necessary—it doesn't seem to add much value since we checkArgument for many other functions :o

Reply:
I see. Actually my bad, checkArgument isn't even in the Date class. will update

Reply:
Updated UML